modules/alloy/src/main/ts/ephox/alloy/positioning/mode/NodeAnchor.ts

```diff
@@ -10,15 +11,16 @@ import * as AnchorLayouts from './AnchorLayouts';
 import * as ContainerOffsets from './ContainerOffsets';
 import * as ContentAnchorCommon from './ContentAnchorCommon';
 
-const placement = (component: AlloyComponent, anchorInfo: NodeAnchor, origin: Origins.OriginAdt): Optional<Anchoring> => {
+const placement = (component: AlloyComponent, anchorInfo: NodeAnchor, origin: Origins.OriginAdt, getBounds: Optional<() => Bounds>): Optional<Anchoring> => {
```

Comment:
This stuff is notoriously difficult to test, because you feel like you are just asserting a value that could be wrong anyway, but you can write simpler property tests for it. If you can identify some invariant that should hold (e.g. box dimensions can never exceed something), then you can test that invariant without actually having to just calculate where it should go.

I thought I'd done this somewhere, but I can't find it currently.

Reply:
Was it this you're thinking of? https://github.com/tinymce/tinymce/blob/417779b3655c97a21789a23a5c0dcb489c239da2/modules/alloy/src/test/ts/atomic/position/view/BounderCalcRepositionTest.ts#L7